### PR TITLE
eating refactor (#3733) + bugfixes

### PR DESCRIPTION
**Pull request by VampyrBytes**
Ok, messed up resolving the merge conflicts (first try), so doing a
fresh pr. This turned out to be a good thing, as after I put the changes
back I tested again, and found 2 bugs (not due to my code)

Firstly, the organ refactor had introduced a bug when taking liver
damage from being drunk. It wasn't getting the liver correctly, leaving
it with null, and was then trying to damage it.

The second was related to the amount you took in when drinking. @Fox-McCloud in [this](http://nanotrasen.se/phpBB3/viewtopic.php?f=12&t=7012&p=55943&hilit=shitfaced#p55804) post you said that glasses gave you 5 units and bottles 10. You were basing this on the amount_per_transfer_from_this var, which would make sense. However, that's not how it was working. The first mouthful you took would always be 5. After that the amount you took in would be one fifth of what was left in the container. So drinking out of a bottle that held 100 units, the first mouthful would be 5, the second would be 19(95/5) the third 15(76/5) down to a minimum of 5.

It is now set so the amount you take = amount_per_transfer_from_this (with a minimum of 5 in case any aren't set)

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
@VampyrBytes 

I know what the code says---and it's...really weird how it's written, but I was basing my experiences on how it *actually* functions, in game. I literally have no idea how it's doing the 5 units and 10 units things---but it was, in my testing. It's ...so strange.

**Comment by VampyrBytes** [comment]
Hmm, that's weird. It was during testing that I found it... I spotted bitesize at 15 so checked the reagents left in a bottle of gin and found them to be 76. Which is when i then tracked down that weird bit of code. Perhaps something I did, or something someone did between your testnig and now reenabled it.
Either way, the strange code is now gone and it's working as intended in this pr

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -18,19 +18,20 @@
...
-		usr.unEquip(src)	//so icons update :[
-
-		if(trash)
-			if(ispath(trash,/obj/item))
-				var/obj/item/TrashItem = new trash(usr)
-				usr.put_in_hands(TrashItem)
-			else if(istype(trash,/obj/item))
-				usr.put_in_hands(trash)
-		qdel(src)
+	spawn(0)
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: what's this spawn for, out of curiosity?

**Comment by Fox-McCloud** [comment]
Also, to make sure, here.

Swallowing a pill would proc `INGEST` and transfer the reagent, instantly, to someone, correct?

Likewise, apply a pill would proc `TOUCH` (based on the volume of the patch) and only transfer half of the reagents, correct? (I saw that you included the transfer efficiency thing, but moved it up the tree).

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (VampyrBytes) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: @Fox-McCloud It's to counteract the spawn(0) in the consume proc (which was in  the original code, and I wasn't sure about taking out). Without it the code below, including the qdel, runs before the reagents have been transferred

**Comment by VampyrBytes** [comment]
Yeah, I moved it up the tree to save having to snowflake in consume. I also gave food the apply_type, which for everything except pills is ingest, and is touch for pills, so that all works in the same way. It also means that if something else comes along that has a different apply_type or transfer efficiency there wil be no need for snowflaking them either

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Alright, thanks.

**Comment by Fox-McCloud** [comment]
@core0verload 

Tagging you because I know you're refactoring growns and stuff on TG and this may interest you, as well.

**Pull request merged**